pkg/scalers/splunk_o11y_scaler.go

```diff
@@ -0,0 +1,264 @@
+	logger := InitializeLogger(config, "splunk_o11y_scaler")
+
+	meta, err := parseSplunkO11yMetadata(config, logger)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Splunk metadata: %w", err)
+	}
+
+	apiClient, err := newSplunkO11yConnection(ctx, meta, config)
+	if err != nil {
+		return nil, fmt.Errorf("error establishing Splunk Observability Cloud connection: %w", err)
+	}
+
+	return &splunkO11yScaler{
+		metadata:  meta,
+		apiClient: apiClient,
+		logger:    logger,
+	}, nil
+}
+
+func parseSplunkO11yMetadata(config *scalersconfig.ScalerConfig, logger logr.Logger) (*splunkO11yMetadata, error) {
+	meta := splunkO11yMetadata{}
+
+	// query
+	if query, ok := config.TriggerMetadata["query"]; ok {
+		meta.query = query
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	// metric name
+	if metricName, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = GenerateMetricNameWithIndex(config.TriggerIndex, kedautil.NormalizeString(fmt.Sprintf("signalfx-%s", metricName)))
+	} else {
+		return nil, fmt.Errorf("no metric name given")
+	}
+
+	// queryValue
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, fmt.Errorf("queryValue parsing error %w", err)
+		}
+		meta.queryValue = queryValue
+	} else {
+		if config.AsMetricSource {
+			meta.queryValue = 0
+		} else {
+			return nil, fmt.Errorf("no queryValue given")
+		}
+	}
+
+	// activationQueryValue
+	meta.activationQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
+		activationQueryValue, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, fmt.Errorf("queryValue parsing error %w", err)
+		}
+		meta.activationQueryValue = activationQueryValue
+	}
+
+	// queryAggregator
+	if val, ok := config.TriggerMetadata["queryAggregator"]; ok && val != "" {
+		queryAggregator := strings.ToLower(val)
+		switch queryAggregator {
+		case "max", "min", "avg":
+			meta.queryAggegrator = queryAggregator
+		default:
+			return nil, fmt.Errorf("queryAggregator value %s has to be one of 'max', 'min', or 'avg'.", queryAggregator)
+		}
+	} else {
+		meta.queryAggegrator = ""
+	}
+
+	// accessToken
+	if accessToken, ok := config.AuthParams["accessToken"]; ok {
+		// cleanedAccessToken := strings.ReplaceAll(accessToken, "\n", "")
+		// cleanedAccessToken = strings.ReplaceAll(accessToken, "\r", "")
+		meta.accessToken = accessToken
+	} else {
+		return nil, fmt.Errorf("no accessToken given")
+	}
+
+	// realm
+	if realm, ok := config.TriggerMetadata["realm"]; ok {
+		meta.realm = realm
+	} else {
+		return nil, fmt.Errorf("no realm given")
+	}
+	logger.Info("Splunk Realm -> %s\n", meta.realm)
+
+	// Debug TODO check
+	meta.vType = v2.ValueMetricType
+
+	return &meta, nil
+}
+
+func newSplunkO11yConnection(ctx context.Context, meta *splunkO11yMetadata, config *scalersconfig.ScalerConfig) (*signalflow.Client, error) {
```

Comment:
`config` seems to be unused in this function. Can you remove the argument, please?

Reply:
Done.